filip/clients/ngsi_v2/cb.py

```diff
@@ -1717,6 +1720,54 @@ def patch_entity(self,
         if update_needed:
             self.update_entity(update_entity)
 
+    def _subscription_dicts_are_equal(self, first: dict, second: dict):
+        """
+        Check if two dictionaries and all sub-dictionaries are equal.
+        Logs a warning if the keys are not equal, but ignores the
+        comparison of such keys.
+
+        Args:
+            first dict: Dictionary of first subscription
+            second dict: Dictionary of second subscription
+
+        Returns:
+            True if equal, else False
+        """
+
+        def _value_is_not_none(value):
+            """
+            Recursive function to check if a value equals none.
+            If the value is a dict and any value of the dict is not none,
+            the value is not none.
+            If the value is a list and any item is not none, the value is not none.
+            If it's neither dict nore list, bool is used.
+            """
+            if isinstance(value, dict):
+                return any([_value_is_not_none(value=_v) for _v in value.values()])
+            if isinstance(value, list):
+                return any([_value_is_not_none(value=_v)for _v in value])
+            else:
+                return bool(v)
+        if first.keys() != second.keys():
+            logger.warning(
+                "Subscriptions contain a different set of fields. "
+                "Only comparing to new fields of the new one."
+            )
+        for k, v in first.items():
+            ex_value = second.get(k, None)
+            if isinstance(v, dict) and isinstance(ex_value, dict):
+                equal = self._subscription_dicts_are_equal(v, ex_value)
+                if equal:
+                    continue
+                else:
+                    return False
+            if not _value_is_not_none(v) and not _value_is_not_none(ex_value):
```

Comment:
Should it be "if not _value_is_not_none(v) **or** not _value_is_not_none(ex_value)" here? Otherwise the none existing keys will not be ignored.

Reply:
Mhm, depends, or? If I have an old subscription with key `e = None`, I want the new key `e =  5` to be an update. Using `or` would lead to `None == 5 -> true` ?!

Reply:
That's true. A `None` normally indicates that the parameter is unset and the default value is used. So I think it would be ok if this function gives a warning or even returns these unequal keys as results? How to deal with these keys will then be the job of specific use case.

Reply:
Ok! Do you want to add a warning yourself and make the changes? Or should I do that?

Reply:
I will do it ;)